app/controllers/wpcc/your_contributions_controller.rb

```diff
@@ -3,17 +3,15 @@
 module Wpcc
   class YourContributionsController < EngineController
     protect_from_forgery
+    after_action :store_eligible_salary, only: :new
```

Comment:
IMHO, if it is being used in one place is not the point. The point, the view is not using the eligible salary in the session at all. Since it is not part of the new view and the action itself it makes more sense to me to be on a filter.